File: examples/python/models/Cube.py
import math
from .ShapeInterface import ShapeInterface
from .ThreeDimensionalShapeInterface import ThreeDimensionalShapeInterface
from .ManageShapeInterface import ManageShapeInterface

class Cube(ShapeInterface, ThreeDimensionalShapeInterface, ManageShapeInterface):
    def __init__(self, length: int):
        self.length = length

    def area(self)->float:
        return 6 * (self.length**2)

    def volume(self)->float:
        return self.length**3

    def calculate(self) -> float:
        return self.area()

File: examples/csharp/Model/Cube.cs
using System;

namespace Model
{
    public class Cube: IShape, IThreeDimensionalShape, IManageShape
    {
        public int Length { get; private set; }

        public Cube(int length)
        {
            this.Length = length;
        }

        public double Area()
        {
            return 6 * Math.Pow(this.Length, 2);
        }

        public double Volume()
        {
            return Math.Pow(this.Length, 3);
        }

        public double Calculate()
        {
            return this.Area();
        }
    }
}
